core/modules.rs

```diff
@@ -521,6 +734,65 @@ impl ModuleMap {
     }
   }
 
+  fn new_json_module(
+    &mut self,
+    scope: &mut v8::HandleScope,
+    name: &str,
+    source: &str,
+  ) -> Result<ModuleId, Error> {
+    let name_str = v8::String::new(scope, name).unwrap();
+    let source_str = v8::String::new(scope, source).unwrap();
+
+    let tc_scope = &mut v8::TryCatch::new(scope);
+
+    let parsed_json = match v8::json::parse(tc_scope, source_str) {
+      Some(parsed_json) => parsed_json,
+      None => {
+        let message = v8::String::new(
+          tc_scope,
+          &format!("\"{}\" is not a valid JSON.", name),
+        )
+        .unwrap();
```

Comment:
In this case it might be worth to run it through `serde_json::from_string()` to get a better error pointing to faulty place in JSON